src/common/components/repositories-home/index.js

```diff
@@ -29,7 +29,7 @@ class RepositoriesHome extends Component {
       }
 
       snaps.snaps.forEach((snap) => {
-        this.props.dispatch(fetchBuilds(snap.git_repository_url, snap.self_link));
+        this.props.fetchBuilds(snap.git_repository_url, snap.self_link);
```

Comment:
`fetchBuilds` could just take snap as parameter and extract repo URL and self link from it, but that's fine too.